tests/e2e/bot/factory.go

```diff
@@ -0,0 +1,234 @@
+	partnerPlugin *partnerplugin.PartnerPlugin,
+	services []CMService,
+	out io.Writer,
+) (*Bot, chan error, error) {
+	key, err := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate key: %w", err)
+	}
+
+	ownerAddr := crypto.PubkeyToAddress(key.PublicKey)
+
+	cmAccountAddress, _, err := f.networkClient.CreateCMAccount(ctx, key)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create CM account: %w", err)
+	}
+
+	if err := f.networkClient.Transfer(ctx, f.networkClient.PrefundedKeys()[0], ownerAddr, e2eCommon.DefaultCMAccountOwnerFunds); err != nil {
+		return nil, nil, fmt.Errorf("failed to transfer funds to cm account owner: %w", err)
+	}
+
+	if err := f.networkClient.AddBotToCMAccount(ctx, cmAccountAddress, key, ownerAddr); err != nil {
+		return nil, nil, fmt.Errorf("failed to add bot to CM account: %w", err)
+	}
+
+	for _, service := range services {
+		if err := f.networkClient.AddCMService(ctx, cmAccountAddress, key, service.Name, service.Fee); err != nil {
+			return nil, nil, fmt.Errorf("failed to add %s service to CM account: %w", service.Name, err)
+		}
+	}
+
+	port := 0
+	if enableRPCServer {
+		port, err = f.resourceManagerSession.GetNetworkPort()
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to get free port: %w", err)
+		}
+	}
+
+	botDir := path.Join(f.dir, strconv.Itoa(len(f.bots)))
+
+	config := config.UnparsedConfig{
+		DeveloperMode:                       true,
+		BotKey:                              hex.EncodeToString(crypto.FromECDSA(key)),
+		CMAccountAddress:                    cmAccountAddress.Hex(),
+		ChainRPCURL:                         f.networkClient.ChainRPCURL(),
+		BookingTokenAddress:                 f.networkClient.BookingTokenContractAddress().Hex(),
+		NetworkFeeRecipientBotAddress:       f.matrix.NetworkFeeRecipientBotAddress().Hex(),
+		NetworkFeeRecipientCMAccountAddress: f.matrix.NetworkFeeRecipientCMAccountAddress().Hex(),
+		ChequeExpirationTime:                3600 * 24 * 30 * 7, // 7 months
+		MinChequeDurationUntilExpiration:    3600 * 24 * 30 * 6, // 6 months
+		CashInPeriod:                        3600,               // 1h
+		ResponseTimeout:                     30000,              // 30s
+		PartnerPlugin: config.PartnerPluginConfig{
+			Enabled:     partnerPlugin != nil,
+			Host:        partnerPlugin.Host(),
+			Unencrypted: true,
+		},
+		RPCServer: config.RPCServerConfig{
+			Enabled:     enableRPCServer,
+			Port:        uint64(port),
+			Unencrypted: true,
+		},
+		Matrix: config.UnparsedMatrixConfig{Host: f.matrix.Host().String()},
+		DB: config.UnparsedSQLiteDBConfig{
+			DBPath:         path.Join(botDir, "db"),
+			MigrationsPath: f.migrationsPath,
+		},
+		Tracing: config.TracingConfig{Enabled: false},
+	}
+
+	if err := os.RemoveAll(botDir); err != nil {
+		return nil, nil, fmt.Errorf("failed to remove bot data dir: %w", err)
+	}
+
+	if err := os.MkdirAll(botDir, 0o644); err != nil {
```

Comment:
Execute bit is missing in the oct permissions for the directory. Is that intended?

Reply:
outdated